tests/utils/test_file.py

```diff
@@ -1,84 +1,53 @@
 import pytest
 import os
 import shutil
-from buildtest.utils.file import (
-    is_dir,
-    is_file,
-    create_file,
-    create_dir,
-    walk_tree,
-)
+import uuid
+from buildtest.utils.file import is_dir, is_file, create_dir, walk_tree, resolve_path
 from buildtest.exceptions import BuildTestError
 
 here = os.path.dirname(os.path.abspath(__file__))
 
 
-@pytest.mark.xfail(
-    reason="Test expected to fail when checking an obscure directory path",
-    raises=BuildTestError,
-)
 def test_checking_directory():
-    is_dir("/xxXXxxyyYYyyYYyyzzZZzzZZzZZZz")
+    dirname = str(uuid.uuid4())
+    assert not is_dir(dirname)
 
 
 @pytest.mark.xfail(
     reason="Test expected to fail when checking an obscure file", raises=BuildTestError
 )
 def test_checking_file():
-    is_file("/xXXXxxXXYyyyYYYyyYzZZZzZZZ")
-
-
-def test_directory_expansion():
-    is_dir("$HOME")
-    is_dir("~")
-
-    assert True is os.path.isdir(os.path.expanduser("~"))
-    assert True is os.path.isdir(os.path.expandvars("$HOME"))
+    file_name = str(uuid.uuid4())
+    assert not is_file(file_name)
 
+    file1 = "~/.profile"
+    file2 = "$HOME/.profile"
 
-def test_check_file():
-    is_file("~/.profile")
-    is_file("$HOME/.profile")
+    assert is_file(file1)
+    assert is_file(file2)
 
-    assert True is os.path.isfile(os.path.expanduser("~/.profile"))
-    assert True is os.path.isfile(os.path.expandvars("$HOME/.profile"))
-
-
-def test_create_file():
-    create_file("/tmp/a.txt")
-    create_file("$HOME/a.txt")
-    create_file("~/b.txt")
-
-    assert True is os.path.isfile("/tmp/a.txt")
-    # checking variable expansion
-    assert True is os.path.isfile(os.path.expandvars("$HOME/a.txt"))
-    # checking ~ expansion
-    assert True is os.path.isfile(os.path.expanduser("~/b.txt"))
-
-    os.remove("/tmp/a.txt")
-    os.remove(os.path.expandvars("$HOME/a.txt"))
-    os.remove(os.path.expanduser("~/b.txt"))
 
+def test_directory_expansion():
+    dir1 = "$HOME"
+    dir2 = "~"
 
-@pytest.mark.xfail(
-    reason="Expected Failure in creating file because lack of permission",
-    raises=OSError,
-)
-def test_fail_create_file():
-    create_file("/etc/a.txt")
+    assert is_dir(dir1)
+    assert is_dir(dir2)
 
 
 def test_create_dir():
-    create_dir("$HOME/a/b/c")
-    create_dir("~/x/y/z")
-    assert True is os.path.isdir(os.path.expandvars("$HOME/a/b/c"))
-    assert True is os.path.isdir(os.path.expanduser("~/x/y/z"))
+    dir1 = "$HOME/a/b/c"
+    dir2 = "~/x/y/z"
+    create_dir(dir1)
```

Comment:
Same question as above. It's generally good practice to use os.path.sep instead of hard coding like that.

Reply:
i used ``os.path.join`` i guess the purpose of these two tests was to check shell and user expansion
with recursive directory creation. Change be seen in https://github.com/HPC-buildtest/buildtest-framework/pull/270/commits/2a8d6749580f2abad0c3e4c65a85b5a3d9e580e3#diff-a42582ff8b56b5783fdd220f2eb5ffab